Don't build with Atomics support by default

saghul opened the pull request:
It's not currently implemented for Windows which makes the bytecode generated by qjsc not portable.

bnoordhuis commented:
macos-asan bots fails to build with:
```
quickjs.c:11528:17: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
                sprintf(q, "%d", p);
```
Which, okay, isn't wrong but not related to sanitizer support. I'm removing the bot for now so I can move forward, will revisit later.

saghul committed:
Don't build with Atomics support by default

It's not currently implemented for Windows which makes the bytecode
generated by qjsc not portable.
